### PR TITLE
Feature/5/importer with load dependencies

### LINKED ISSUE #4: Implement loading of asset dependency graphs
The `rpc_loader` should support automatically loading an asset's dependency graph before making the load visible to users.

The `load_deps` field in the `AssetMetadata` already exists, along with `get_metadata_with_dependencies` in the RPC server. This should be sufficient to implement the logic in the `loader`.

The whole asset graph should be kept uncommitted in the storage until all assets can be committed.

### PR DESCRIPTION
**Pull request by azriel91**
Closes #4 and #5.

This is based on top of #12, so might want to merge that first and I can rebase.

Feel free to ask me to explain if the code isn't clear enough (am too tired right now).

### CODE REVIEW

**Comment by kabergstrom** [comment]
Thank you for this! It will be a good basic implementation of loading dependencies 😁

Just one thing that I think should be changed: an asset can go through the ProcessingQueue even though its dependency is not loaded yet, which is probably rather unexpected. You need to move the waiting code from precommit to RequestDependencies to fix this.

This PR obviously doesn't address all potential cases of dependency graphs but that's fine, I'll make a separate issue to explore the problem space in more detail so we can get more collective brainpower on it. 😉

I'll merge once you've rebased and fixed the ordering issue mentioned.

File: loader/src/rpc_loader.rs [hunk 1]
```diff
@@ -523,6 +544,50 @@ fn process_load_states(
...
+                        .expect("in `LoadedPrecommit` state but asset_type is None");
+
+                    // Ensure dependencies are committed before committing this asset.
+                    // load is an AssetLoad.
+                    let asset_dependencies_committed = {
+                        let asset_id = value.asset_id;
+                        let asset_metadata = metadata.get(&asset_id).unwrap_or_else(|| {
+                            panic!("Expected metadata for asset `{:?}` to exist.", asset_id)
+                        });
+                        asset_metadata.load_deps.iter().fold(
```
[the remark this answers is not in the document]
Comment (azriel91) [reply loader/src/rpc_loader.rs]: it does :stuck_out_tongue:

**Comment by azriel91** [comment]
Cool, I *think* I did that correctly, thanks for the help!

File: daemon/src/daemon.rs [hunk 2]
```diff
@@ -24,7 +24,7 @@ pub struct AssetDaemon {
 impl Default for AssetDaemon {
     fn default() -> Self {
         Self {
-            db_dir: PathBuf::from(".amethyst"),
+            db_dir: PathBuf::from(".db"),
```
Comment (kabergstrom) [inline daemon/src/daemon.rs]: I think `.db` is a bit too generic? Maybe something about assets or amethyst would be good.

File: loader/src/rpc_loader.rs [hunk 3]
```diff
@@ -507,12 +527,56 @@ fn process_load_states(
...
+                    LoadState::WaitingForDependencies
+                }
+                LoadState::WaitingForDependencies => {
+                    let asset_id = value.asset_id;
+                    let asset_metadata = metadata.get(&asset_id).unwrap_or_else(|| {
+                        panic!("Expected metadata for asset `{:?}` to exist.", asset_id)
+                    });
+
+                    // Ensure dependencies are committed before continuing to load this asset.
+                    let asset_dependencies_committed = {
```
Comment (kabergstrom) [inline loader/src/rpc_loader.rs]: what's with the enclosing scope? Is it some convention I'm unaware of?

**Pull request merged**